extensions/wsc/src/main/java/org/mule/extension/ws/internal/metadata/NodeElementResolver.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.extension.ws.internal.metadata;
+
+import static java.lang.String.format;
+import static org.mule.metadata.api.builder.BaseTypeBuilder.create;
+import static org.mule.metadata.api.model.MetadataFormat.XML;
+import static org.mule.runtime.api.metadata.resolving.FailureCode.UNKNOWN;
+import org.mule.metadata.api.model.BooleanType;
+import org.mule.metadata.api.model.DateTimeType;
+import org.mule.metadata.api.model.DateType;
+import org.mule.metadata.api.model.MetadataType;
+import org.mule.metadata.api.model.NumberType;
+import org.mule.metadata.api.model.impl.DefaultStringType;
+import org.mule.metadata.xml.XmlTypeLoader;
+import org.mule.runtime.api.connection.ConnectionException;
+import org.mule.runtime.api.metadata.MetadataContext;
+import org.mule.runtime.api.metadata.MetadataResolvingException;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.wsdl.Part;
+
+/**
+ * Base class for metadata resolvers that resolve dynamic metadata of XML node elements.
+ *
+ * @since 4.0
+ */
+public abstract class NodeElementResolver extends BaseWscResolver {
+
+  private static final Map<String, MetadataType> TYPES = new LinkedHashMap<>();
+  private static final MetadataType NULL_TYPE = create(XML).nullType().build();
+
+  protected final TypeResolverDelegate delegate;
+
+  protected NodeElementResolver(TypeResolverDelegate delegate) {
+    this.delegate = delegate;
+  }
+
+  static {
+    DefaultStringType stringType = create(XML).stringType().build();
+    BooleanType booleanType = create(XML).booleanType().build();
```

Comment:
isn't it a problem that none of these types have a typeId? If so, shouldn't use use different numberTypes for int than long?